privacyidea/lib/policy.py

```diff
@@ -706,8 +706,10 @@ def match_policies(self, name: Optional[str] = None, scope: Optional[str] = None
                 [p.get('name') for p in reduced_policies]))
 
         if audit_data is not None:
+            audit_policies = set(audit_data.get("policies", []))
```

Comment:
we have this `add_policy()` function in the Audit class, maybe we can use that:
https://github.com/privacyidea/privacyidea/blob/5510737af4ace5b2f6405a06fd4a4c3b90b88d07/privacyidea/lib/auditmodules/base.py#L198

Reply:
Would it make sense to pass the audit object instead of just the `audit_data` dictionary?